tests/Unit/Core/Context/LanguageContextBuilderTest.php

```diff
@@ -89,6 +89,7 @@ public function testBuildLegacyContext(): void
         $builder->setLanguageId($objectModelLanguage->getId());
 
         $contextManagerMock->expects(static::once())->method('setLanguage')->with($objectModelLanguage);
+        $contextManagerMock->expects(static::once())->method('setCurrentLocale')->with($this->mockObjectModelLanguageRepository($objectModelLanguage));
```

Comment:
I think you need to mock the language object only once and store it in a variable, you can then pass on this line and on line 87